src/main/java/network/brightspots/rcv/TabulatorSession.java

```diff
@@ -103,30 +105,40 @@ void tabulate() {
     Logger.log(Level.INFO, "Starting tabulation session...");
     ContestConfig config = ContestConfig.loadContestConfig(configPath);
     if (config != null && config.validate() && setUpLogging(config)) {
-      if (config.isSequentialMultiSeatEnabled()) {
-        int numWinners = config.getNumberOfWinners();
-        // temporarily set config to single-seat so we can run sequential elections
-        config.setNumberOfWinners(1);
-        while (config.getSequentialWinners().size() < numWinners) {
-          Set<String> newWinnerSet = runTabulationForConfig(config);
-          assert (newWinnerSet.size() == 1);
-          String newWinner = (String) newWinnerSet.toArray()[0];
-          config.setCandidateExclusionStatus(newWinner, true);
-          config.addSequentialWinner(newWinner);
-        }
-        // revert config to original state
-        config.setNumberOfWinners(numWinners);
-        for (String winner : config.getSequentialWinners()) {
-          config.setCandidateExclusionStatus(winner, false);
+        if (config.isSequentialMultiSeatEnabled()) {
+          int numWinners = config.getNumberOfWinners();
+          // temporarily set config to single-seat so we can run sequential elections
+          config.setNumberOfWinners(1);
+          while (config.getSequentialWinners().size() < numWinners) {
+            // Read cast vote records and precinct IDs from CVR files
+            List<CastVoteRecord> castVoteRecords = parseCastVoteRecords(config, precinctIDs);
```

Comment:
This same block seems to appear in both parts of the if statement (if and else); maybe it's possible to factor outside of it? Ignore me if this is wrong.

Reply:
Right, that's what I was slacking with Louis about before.  This needs to be called in both scenarios but for sequential it happens multiple times.

Reply:
I don't quite understand why you pulled `parseCastVoteRecords` out of `runTabulationForConfig`. If you left it in there, you could avoid this duplicated logic. If the issue is that you want to break out of the sequential loop, you could do that by just testing whether it returns an empty Set... or you could make it throw an exception that you catch.

Reply:
Yeah, I actually tried those variations and landed on this because of the way it more clearly shows the  castVoteRecord parsing as a prerequisite to tabulating.  The high level flow and failure paths made more sense.